src/Compilers/CSharp/Portable/Symbols/SourceEmbeddedAttributeSymbol.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.Cci;
+using Roslyn.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Runtime.InteropServices;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Emit;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Represents a compiler generated and embedded attribute type.
+    /// This type has the following properties:
+    /// 1) It is non-generic, sealed, non-static class.
+    /// 2) It implements System.Attribute
+    /// 3) It has Microsoft.CodeAnalysis.EmbdeddedAttribute
+    /// 4) It has System.Runtime.CompilerServices.CompilerGeneratedAttribute
+    /// 5) It has a parameter-less constructor
+    /// </summary>
+    internal sealed class SourceEmbeddedAttributeSymbol : NamedTypeSymbol
+    {
+        private string _name;
+
+        private NamedTypeSymbol _baseType;
+        private MissingNamespaceSymbol _namespace;
+        private ImmutableArray<Symbol> _members;
+        private ModuleSymbol _module;
+
+        public SourceEmbeddedAttributeSymbol(WellKnownType wellKnownType, CSharpCompilation compilation)
+        {
+            var nameParts = WellKnownTypes.GetMetadataName(wellKnownType).Split('.');
+            Debug.Assert(nameParts.Length > 1, "Attribute cannot be in global namespace?");
```

Comment:
I'd say yes, since the code below operates on that fact.

---
In reply to: [113527748](https://github.com/dotnet/roslyn/pull/18715#discussion_r113527748) [](ancestors = 113527748)